api/app/controllers/transcript.controller.js

```diff
@@ -1,23 +1,61 @@
+const authConfig = require("../config/auth.config.js");
 const db = require("../models");
 const Transcript = db.transcripts;
 const Op = db.Sequelize.Op;
+const crypto = require('crypto');
 
 // Create and Save a new Transcript
 exports.create = (req, res) => {
   // Validate request
   if (!req.body.content) {
-    //FIXME: Include original content check in if condition
     res.status(400).send({
-      message: "Content can not be empty!"
+      message: "Content cannot be empty!"
+    });
+    return;
+  }
+  
+  else if (!req.body.originalContent) {
+    res.status(400).send({
+      message: "Original Content cannot be empty!"
+    });
+    return;
+  }
+
+  // add authentication token to prevent dos attacks
+  else if (!req.body.authToken) {
+    res.status(400).send({
+      message: "Auth token cannot be empty!"
     });
     return;
   }
 
+
+  const verifyToken = (token) => {
+
+    return token === authConfig.TOKEN;
+  }
+
+  const generateHash = () => {
+
+    const oc = req.body.originalContent
+    const hashParams = oc.title + oc.media + oc.date
+    const transcriptHash = crypto.createHash('sha256').update(hashParams).digest('base64');
+
+    return transcriptHash;
+  }
+
+
+  if (!verifyToken(req.body.authToken)) {
+    res.status(400).send({
+      message: "Auth token is invalid"
+    });
+    return;
+  }
   // Create a Transcript
   const transcript = {
-    // We have to add title because for some reason having just content makes an update to an existing record insted of inserting a new one
     originalContent: req.body.originalContent,
-    content: req.body.content
+    content:req.body.originalContent,
```

Comment:
content and originalContent both refer to `req.body.originalContent` why is that?

Reply:
nice catch. going to fix that...